docs/markdown/Qt5-module.md

```diff
@@ -3,27 +3,97 @@
 The Qt5 module provides tools to automatically deal with the various
 tools and steps required for Qt. The module has two methods.
 
+## compile_resources
+
+*New in 0.59.0*
+
+Compiles Qt's resources files (.qpa) into c++ files for compilation.
+
+It takes no positional arguments, and the following keyword arguments:
+  - `name` (string | empty): if provided a single .cpp file will be generated,
+    and the output of all qpa files will be combined in this file, otherwise
+    each qpa file be written to it's own cpp file.
+  - `sources` (File | string)[]: A list of sources to be transpiled. Required,
+    must have at least one source
+  - `extra_args` string[]: Extra arguments to pass directly to `qt-rcc`
+  - `method` string: The method to use to detect qt, see `dependency()` for more
+    information.
+
+## compile_ui
+
+*New in 0.59.0*
+
+Compiles Qt's ui files (.ui) into header files.
+
+It takes no positional arguments, and the following keyword arguments:
+  - `sources` (File | string)[]: A list of sources to be transpiled. Required,
+    must have at least one source
+  - `extra_args` string[]: Extra arguments to pass directly to `qt-uic`
+  - `method` string: The method to use to detect qt, see `dependency()` for more
+    information.
+
+## compile_moc
+
+*New in 0.59.0*
+
+Compiles Qt's moc files (.moc) into header and/or source files. At least one of
+the keyword arguments `headers` and `sources` must be provided.
+
+It takes no positional arguments, and the following keyword arguments:
+  - `sources` (File | string)[]: A list of sources to be transpiled int .cpp files
+  - `headers` (File | string)[]: A list of sources to be transpiled int .h files
```

Comment:
```suggestion
  - `sources` (File | string)[]: A list of sources to be transpiled into moc_*.cpp files
  - `headers` (File | string)[]: A list of headers to be transpiled into moc_*.cpp files
```

Moc always generates cpp files, that then need to be compiled.

I think those are both compiled into .cpp files though?

Reply:
The existing code transforms the headers .moc files, and the sources into `.cpp` files

Reply:
Interesting, I would have maybe expected both, not just a moc file. Especially since if I actually run it on a project of mine, it does generate a moc_*.cpp file?

Reply:
Wierd.... the headers are transpiled into `.cpp` files and the sources into `.moc` files... Does that seem right?

Reply:
Okay, I've changed this to:
```markdown
  - `sources` (File | string)[]: A list of sources to be transpiled int .moc
    files for manual inclusion.
  - `headers` (File | string)[]: A list of headers to be transpiled int .cpp files
```

Reply:
Okay, to summarize the matrix discussion, source files seem to be compiled to .moc, headers to .cpp. The former you need to manually include in your sources, the latter should be added to another target to be compiled and linked.

Reply:
Why "int" instead of "into" btw?

Reply:
lol, typo + copy paste fail